cnudie/util.py

```diff
@@ -414,6 +416,121 @@ def find_changed_component(
     )
 
 
+def format_component_diff(
+    component_diff: ComponentDiff,
+    to_component_name: str,
+    delivery_dashboard_url: str | None
+):
+    if delivery_dashboard_url:
+        dashboard_view_diff = (
+            f'{delivery_dashboard_url}/#/component?name='
+            f'{to_component_name}&view=diff'
+        )
+        bom_diff_header = f'## <a href="{dashboard_view_diff}">BoM Diff</a>\n'
+    else:
+        bom_diff_header = '## BoM Diff\n'
+
+    added_components = [
+        f'\U00002795 {component.name} (v{component.version})'
+        for component in component_diff.cidentities_only_right
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    removed_components = [
+        f'\U00002796 {component.name} (v{component.version})'
+        for component in component_diff.cidentities_only_left
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    changed_components = [
+        f'\U00002699 {name}: {old_component.version} → {new_component.version}'
+        for name in component_diff.names_version_changed
```

Comment:
What's the benefit of iterating `names_version_changed` first and then searching the corresponding entry in `cpairs_version_changed` instead of directly only iterating `cpairs_version_changed`?